Assets/HoloToolkit/Input/Scripts/InputManager.cs

```diff
@@ -228,30 +235,35 @@ public void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<
             }
 
             // Use focused object when OverrideFocusedObject is null.
-            GameObject focusedObject = (OverrideFocusedObject == null) ? GazeManager.Instance.HitObject : OverrideFocusedObject;
+            GameObject focusedObject = OverrideFocusedObject ?? GazeManager.Instance.HitObject;
+            GameObject modalInput = modalInputStack.Count > 0 ? modalInputStack.Peek() : null;
+            GameObject fallbackInput = fallbackInputStack.Count > 0 ? fallbackInputStack.Peek() : null;
 
-            // Send the event to global listeners
+            // Send the event to global listeners.
             for (int i = 0; i < globalListeners.Count; i++)
             {
-                // Global listeners should only get events on themselves, as opposed to their hierarchy
-                ExecuteEvents.Execute(globalListeners[i], eventData, eventHandler);
+                // Only execute the global event if it's not already being handled.
```

Comment:
I'm not sure why we would want to do that.

The design of the global listeners was that they will always get input events, no matter what happens and without consideration for anything else that might be handling the event. That's why they are global. Yes, it leaves a possibility for a double registration, but that kind of error would be on the programmer. If you want something that will only be called when nothing else handles it, you should use the fallback handler, which was designed for that purpose. I don't see a strong case for changing this design.

Also, I don't think that what is being done here would actually work, This is only looking at the top of the modal and fallback stacks, so anything else in these stacks will not be considered. It's also not considering hierarchy when doing the if condition that checks if the event is being handled, so it's not really accurate.